### ISSUE TITLE
Thumbnails are not being displayed in Windows File Explorer when using RecipeThumbnailProvider

### ISSUE THREAD

**Issue by kareldonk** [microsoft/Windows-classic-samples#159]
When building and installing the [`RecipeThumbnailProvider`](https://github.com/microsoft/Windows-classic-samples/tree/master/Samples/Win7Samples/winui/shell/appshellintegration/RecipeThumbnailProvider) example it works only for files that are located on a local hard disk drive. It does not work for files that are on a CD/DVDROM drive.

Steps to reproduce:

1. After building the `RecipeThumbnailProvider` sample using Visual Studio, register the resulting DLL file as follows on the command line:

`regsvr32 RecipeThumbnailProvider.dll`

2. There is a [`chocolatechipcookies.recipe`](https://github.com/microsoft/Windows-classic-samples/blob/master/Samples/Win7Samples/winui/shell/appshellintegration/RecipeThumbnailProvider/chocolatechipcookies.recipe) file provided in the example. Place this file in a folder on the PC and in Windows File Explorer, navigate to the file and confirm that you see a thumbnail for the file when you enable thumbnail previews in Windows File Explorer (see screenshot 1).
![screen1](https://user-images.githubusercontent.com/12107205/95623364-83809880-0a4b-11eb-9670-b6d82c8c63e4.png)

3. When the recipe file `chocolatechipcookies.recipe` is burned to a CD/DVDROM disk and placed in the drive, if you open the CD/DVDROM drive in Windows File Explorer, thumbnails are not displayed for the file even when thumbnail previews are activated. See screenshot 2. ![screen2](https://user-images.githubusercontent.com/12107205/95623536-d0fd0580-0a4b-11eb-9278-24f168ab630c.png)

I have attached a ISO disk image with the recipe file in order to make it easier to test. You can easily mount this image as a CDROM drive in a virtual machine to test: [brasero.zip](https://github.com/microsoft/Windows-classic-samples/files/5357049/brasero.zip)

It appears this same problem also exists with other extensions including one I made myself called [WMFPreview](https://github.com/kareldonk/WMFPreview). As soon as the files are on a CD/DVD drive (and some users have also reported Microsoft OneDrive), Windows File Explorer does not display thumbnails for the files. The files have to be copied to a local folder on the hard disk for it to work.

I'm hoping someone from Microsoft can shine more light on this problem ASAP.

**Comment by kareldonk** [comment]
@Karl-Bridge-Microsoft  @oldnewthing  Please help/check. All thumbnail shell extensions appear to be affected.

**Comment by HJay75** [comment]
What kareldonk said! .WMF files will not show thumbnail's in OneDrive they have to be downloaded to the computer to work, which kind of makes using OneDrive to store .WMF flies pointless. Please can we have a fix for this?

**Comment by oldnewthing** [comment]
This isn't a question about the thumbnail sample. It's a question about the policy under which Windows File Explorer chooses to show thumbnails or not. The sample is just fine. 

You can follow the instructions on [Windows developer support](https://developer.microsoft.com/en-us/windows/support/) for support options. (Click "Forums and community".)

**Comment by kareldonk** [comment]
@oldnewthing As far as I can tell there is absolutely zero documentation about the supposed "policy under which Windows File Explorer chooses to show thumbnails or not". If you have any, could you point us to it and also make a note about it in the SDK? It looks more like a bug to me than a feature. I spent hours trying to "debug" both my code and the sample to try to find out what I was missing. The fact that there is apparently  no documentation also didn't help. Why do JPG files show thumbnails for the supposed "slow media" that you mention? If JPG files (and perhaps more, I haven't checked) can have thumbnails, why not lightweight WMF? This does not make sense. 
Do you know anyone on the Windows Shell team that can shed more light on this?

**Comment by oldnewthing** [comment]
Explorer follows the operating system guidance that [offline files are not recalled until the user explicitly takes action on them](https://devblogs.microsoft.com/oldnewthing/20090415-00/?p=18553).

OneDrive downloads a preview of image files (for thumbnail purposes) rather than the full file. This is a special thing that OneDrive does. Explorer isn't directly involved in that.

I'll add a note to the sample that Explorer may choose not to display thumbnails even if a provider is available, based on its own internal policies.

**Comment by kareldonk** [comment]
@oldnewthing Thanks for adding the note. However, it doesn't explain why we also get JPG previews for images on CDROM. I used to also think it had something to do with slow drives, but then it would mean that JPG images on CDROM would also not show previews. The question is, why does Explorer decide to show previews for some files, such as the JPGs, but not other, often smaller files? Is this a bug, some other problem, or perhaps some setting in the preview handlers?

**Comment by oldnewthing** [comment]
I do not know what Explorer's policies are. You can ask the Explorer team. This is the Samples repo; the Explorer team doesn't hang out here.

**Comment by kareldonk** [comment]
@oldnewthing Since this problem affects your sample, I thought you would be able to check why this is happening. Especially since you work for the same company and might have better access to developers at Microsoft.

By the way, if it really is some kind of policy by Explorer, it would also be nice for the details on that to get published in the documentation. It is very unintuitive; not only for developers but also for users who write in to me to complain about this inconsistent behavior. For developers it can cause unnecessary time spent on troubleshooting, as in my case. It would be nice to have the policies clearly stated in the docs, and based on what Explorer makes these decisions, so that developers know when their extensions are working correctly or when there is a problem.

**Comment by oldnewthing** [comment]
I'm working to add this sentence to the documentation:

> Be aware that Explorer may choose not to display a thumbnail even if a provider is available. For example, a file that has been
archived to tape will not be recalled to obtain its thumbnail.

**Comment by kareldonk** [comment]
@oldnewthing Thanks a lot. It would also be helpful if we could know under which exact circumstances Explorer will not display thumbnails. That will be helpful to developers and users and eliminate guesswork on our part. The SDK is incomplete without that important info in my opinion.
For example, for months I thought I was the one doing something wrong until I finally took the time to spend hours on trying to find out where my mistake/bug was (after users kept telling me they couldn't browse their files on CDROM). I spent a lot of time going through the docs and found nothing so I thought I was doing something wrong, perhaps missing some setting or flag. It wasn't until I tried the SDK sample when I saw the same issue there.
